Homework2 Draft1 Submission (#5)
tstang99/HW2-sugary-beverages: report/main.qmd

---

> title: "What is the best way to promote zero calories beverages?"

Reviewer: The report is very well-organized, and the writing is in very good logic and structure. The choice of model is quite reasonable. In formal analysis, I think you can add some more specific interpretations and detailed arguments, some more examples could be used to illustrate the ideas as well. 

The most of comments are my personal thoughts and just for your reference.

format: 
  pdf:
    fontsize: 11pt
editor: visual
bibliography: ../reference/reference.bib
author: Ivy Liu, Zefan Liu, Tom Tang
---

## Introduction

> Zero calorie beverages are healthier compared to sugared beverages because they contain less sugar and significantly fewer calories. Over the past decades, hospitals have been striving to promote zero-calorie beverages for a healthier lifestyle. This study employs five interventions to help increase such promotion. These interventions include a 10% price discount, a price discount combined with messaging that explains the reason for the discount, messaging displaying the caloric content in sugared beverages, messaging indicating the amount of exercise needed to burn off the calories in a sugared beverage, and messaging containing both caloric content and the amount of physical activity needed.

Reviewer 2: ```suggestion
Zero-calorie beverages are healthier than sugared beverages because they contain less sugar and significantly fewer calories. Over the past few decades, hospitals have been striving to promote zero-calorie beverages for a healthier lifestyle. This study employs five interventions to help increase such promotion. These interventions include a 10% price discount, a price discount combined with messaging that explains the reason for the discount, messaging displaying the caloric content in sugared beverages, messaging indicating the amount of exercise needed to burn off the calories in a sugared beverage, and messaging containing both caloric content and the amount of physical activity needed.
```

> The study aims to investigate whether these interventions help encourage people to choose zero calorie beverages over sugared beverages and if such effects are consistent across hospitals. The primary statistical question in this context is: "Is there any association between interventions and the increase in zero calorie beverage consumption or decrease in sugary beverage consumption?" The secondary statistical question is:"Does the effect of intervention differs by site?"

Reviewer 2: ```suggestion
The study aims to investigate whether these interventions help encourage people to choose zero-calorie beverages over sugared beverages and whether such effects are consistent across hospitals. The primary statistical question in this context is: "Is there any association between interventions and the increase in zero calorie beverage consumption or decrease in sugary beverage consumption?" The secondary statistical question is:"Does the effect of intervention differ by site?"
```

> 

Reviewer: I suggest adding one paragraph to talk about the organization for following sections.

> 

Reviewer: The introduction is good, I suggest adding one paragraph to talk about the organization for following sections.

## Data description and summaries

> This is an experimental design where data are collected over 30 weeks, from October 27 to May 23, with a follow-up period of 14 days, gathered from two urban hospitals and one suburban hospital. During this time period, the sales of different types of drinks are recorded.

Reviewer 2: ```suggestion
This is an experimental design where data are collected over 30 weeks, from October 27 to May 23 (it would be clearer if years can be specified),  with a follow-up period of 14 days, involving two urban hospitals and one suburban hospital. During this time period, the sales of different types of drinks are recorded.
```

> This is an experimental design where data are collected over 30 weeks, from October 27 to May 23, with a follow-up period of 14 days, gathered from two urban hospitals and one suburban hospital. During this time period, the sales of different types of drinks are recorded.

Reviewer: I think it's better to emphasize 'quasi-experiment'

> This is an experimental design where data are collected over 30 weeks, from October 27 to May 23, with a follow-up period of 14 days, gathered from two urban hospitals and one suburban hospital. During this time period, the sales of different types of drinks are recorded.

Reviewer: I think it's better to emphasize 'quasi-experiment', and I suggest to provide more interpretations about some of crucial variables. Maybe we can give more specific number of time counts to illustrate the sample size.

| Data Name    | Data Type   | Data Description                       |
|--------------|-------------|----------------------------------------|
| Count        | Integer     | The number of days                     |
| DofW         | Categorical | The day of the week, 7 levels          |
| Site         | Categorical | The location of the hospital, 3 levels |
| Intervention | Categorical | The intervention applied, 9 levels     |
| ZeroCal      | Integer     | The number of zero calorie drink sold  |
| Sugary       | Integer     | The number of sugary drink sold        |
| Juice100     | Integer     | The number of Juice100 sold            |
| Ojuice       | Integer     | The number of Ojuice sold              |
| Sports       | Integer     | The number of sports drink sold        |
| Total        | Continuous  | The sum of all drink sold              |

> 

Reviewer: I'm considering if we should add caption or title for this table.

Reviewer 3: Yes. Tables should "float" rather than running off the page.

> There are missing entries in this dataset, with a significant portion of the missing data attributed to the 'Juice100,' 'Ojuice,' and 'Sports' columns. These columns will not be included in the model because of their irrelevancy, so the missing entries in these columns will not affect the statistical analysis. Additionally, there are seven entirely missing consecutive entries at HF hospital during the follow-up period. This is possible due to the hospital's cafeteria and convenient shop close down for renovation. Apart from these, there are two isolated cases of missing entries. These can be reasonably attributed to temporary closures for public holidays.

Reviewer 2: ```suggestion
There are missing entries in this dataset, with a significant portion of the missing data attributed to the 'Juice100,' 'Ojuice,' and 'Sports' columns. These columns will not be included in the model because of their irrelevance to the study objectives, ensuring the missing entries in these columns do not affect the statistical analysis. Additionally, there are seven entirely missing consecutive entries at HF hospital during the follow-up period. This is possible due to the hospital's cafeteria and convenient shop close down for renovation. Apart from these, there are two isolated cases of missing entries. These can be reasonably attributed to temporary closures for public holidays.
```

Reviewer 3: "Irrelevance" is a very strong statement. I wouldn't say that. Justify your decision.

## Exploratory Data Analysis

```{r, include=FALSE}
library(here)
library(readr)
library(dplyr)
data <- read_csv(here("rawdata/june1data.csv"))
data <- filter(data, Total > 0) 
data <- data[,-c(7,8,9)]
```

> Specific to the statistical questions listed in the introduction, the following exploratory data analysis [@tukey1977exploratory] procedures are suggested. Firstly, in order to address the influence of location on zero-calorie beverage consumption, which is of primary interest to the study, a spaghetti plot is recommended. A spaghetti plot is typically used to demonstrate the change of multiple flows over time and how it varies across the three sites. Meanwhile, it is recommended to use box-plots to approach the primary statistical question concerning the impact of different interventions on zero-calorie and sugary beverage consumption. Box-plots enable a preliminary check of consumption distributions under various interventions. Illustrated below are examples of a spaghetti plot and a box-plots (note that for demonstration purposes, only three interventions are included in the example box-plots).

Reviewer: Small typo in the last sentence: **a** box-plots

Reviewer 3: No need to cite Tukey.

> Specific to the statistical questions listed in the introduction, the following exploratory data analysis [@tukey1977exploratory] procedures are suggested. Firstly, in order to address the influence of location on zero-calorie beverage consumption, which is of primary interest to the study, a spaghetti plot is recommended. A spaghetti plot is typically used to demonstrate the change of multiple flows over time and how it varies across the three sites. Meanwhile, it is recommended to use box-plots to approach the primary statistical question concerning the impact of different interventions on zero-calorie and sugary beverage consumption. Box-plots enable a preliminary check of consumption distributions under various interventions. Illustrated below are examples of a spaghetti plot and a box-plots (note that for demonstration purposes, only three interventions are included in the example box-plots).

Reviewer: Here you use count of bottles to indicate the consumption of the drinks, I was wondering if the raw count could illustrate the relationships of change tendency among different types of beverages, for example, in this way does the increasing of zero-calories mean the decreasing of sugared beverage? I suggest trying to convert the raw count to percentage to see if there are some different findings.

> Specific to the statistical questions listed in the introduction, the following exploratory data analysis [@tukey1977exploratory] procedures are suggested. Firstly, in order to address the influence of location on zero-calorie beverage consumption, which is of primary interest to the study, a spaghetti plot is recommended. A spaghetti plot is typically used to demonstrate the change of multiple flows over time and how it varies across the three sites. Meanwhile, it is recommended to use box-plots to approach the primary statistical question concerning the impact of different interventions on zero-calorie and sugary beverage consumption. Box-plots enable a preliminary check of consumption distributions under various interventions. Illustrated below are examples of a spaghetti plot and a box-plots (note that for demonstration purposes, only three interventions are included in the example box-plots).

Reviewer 2: ```suggestion
Addressing the statistical questions listed in the introduction, the following exploratory data analysis [@tukey1977exploratory] procedures are suggested. Firstly, in order to address the influence of location on zero-calorie beverage consumption, which is of primary interest to the study, a spaghetti plot is recommended. A spaghetti plot is typically used to demonstrate the individual data trajectories over time and equally visualizing its variations across different locations. Meanwhile, it is recommended to use box plots to approach the primary statistical question concerning the impact of different interventions on zero-calorie and sugary beverage consumption. Box plots enable a preliminary check of consumption distributions under various interventions. Illustrated below are examples of a spaghetti plot and a box-plots (note that for demonstration purposes, only three interventions are included in the example box-plots).
```

> ```{r figs, echo=FALSE, fig.width=7,fig.height=4,fig.cap="\\label{fig:figs}Examples of Recommended EDA Plots"}

Reviewer: I suggest using more clear and specific caption for the figures, for now I cannot understand it without reading the details.

library(ggplot2)
library(cowplot)

> p1=ggplot(data=data, aes(x=Count, y=ZeroCal, color=Site))+

Reviewer: I suggest put all library (package) together at the start of the codes, and use more readable name for 'data' here.

  geom_point()+
  geom_line()+
  labs(x="Days of Experiment", y="ZeroCal (bottle)",
       title="The Spaghetti Plot")+
  theme(plot.title=element_text(hjust=0.5),
        legend.position="bottom")

example_data <- data |> filter(Intervention %in% 
                                 c("cal", "dismes", "preint"))

p2=ggplot(data=example_data, aes(x=Intervention, y=ZeroCal))+
  geom_boxplot(outlier.colour="red", outlier.shape=8,
                outlier.size=4)+
  labs(x="Interventions", y="ZeroCal (bottle)",
       title="The Box-plot")+
  theme(plot.title=element_text(hjust=0.5),
        legend.position="bottom")

plot_grid(p1,p2, ncol=2)
```

> Based on the two plots above, it's clear that the time-series data for zero-calorie beverage consumption at site "chop" significantly diverges from the patterns observed at the other two sites, and that the distributions of consumption may vary substantially under different interventions. Additionally, there are two observations. Firstly, the spaghetti plot reveals that except for the beginning period at site "HF", there is no apparent trend at all three sites as the long-term averages seem to be stable. However, strong weekly variations are presented, implying that it is reasonable to include the day of the week (DofW) and to exclude the time (count) in the model. Secondly, the box-plots indicate the presence of many outliers during which the caloric messaging was displayed. However, the clustering nature of these outliers tells valuable information. Therefore, it is advised that outliers be retained in the analysis to reflect the true variability inherent in the intervention's effects.

Reviewer: could you clarify what 'valuable information' that outliers tell?

Reviewer 3: The plots float, they're not "above" so point to them directly.

Reviewer 3: The spaghetti plot is very hard to read. What should I be looking for?

Reviewer 3: I don't understand "outliers be retained to reflect the true variability inherent".

> Based on the two plots above, it's clear that the time-series data for zero-calorie beverage consumption at site "chop" significantly diverges from the patterns observed at the other two sites, and that the distributions of consumption may vary substantially under different interventions. Additionally, there are two observations. Firstly, the spaghetti plot reveals that except for the beginning period at site "HF", there is no apparent trend at all three sites as the long-term averages seem to be stable. However, strong weekly variations are presented, implying that it is reasonable to include the day of the week (DofW) and to exclude the time (count) in the model. Secondly, the box-plots indicate the presence of many outliers during which the caloric messaging was displayed. However, the clustering nature of these outliers tells valuable information. Therefore, it is advised that outliers be retained in the analysis to reflect the true variability inherent in the intervention's effects.

Reviewer 2: >  I suggest writing ".... at site "chop" diverges......" instead as @significantly is mostly paired with statistical significance in formal analysis.

> "....averages seem to be stable.". -  @Stability is not necessarily implied by the absence of apparent trend. An absence of trend might be as a result of inadequate time for trend detection or due to presence of noise in the data. You might want to avoid "overgeneralizing".

> "However, @strong weekly variations...." - Visual inspection does not imply final justification. It might be safer to say such observation calls for testing hypothesis on possible weekly variation which would be based on statistical tests.

> "...the box-plots indicate the presence of many @outliers..." - as a client I might want to know what an outlier is.

Reviewer 2: ```suggestion
Based on the two plots above, it appears that the time-series data for zero-calorie beverage consumption at site 'chop' diverges from the patterns observed at the other two sites, and that the distributions of consumption may vary substantially under different interventions. Additionally, there are two observations. Firstly, the spaghetti plot reveals that except for the beginning period at site 'HF', there is no apparent trend at any of the three sites as the long-term averages seem to be stable. However, strong weekly variations are presented, implying that it is reasonable to include the day of the week (DofW) and to exclude the time (count) in the model. Secondly, the box-plots indicate the presence of many outliers during which the caloric messaging was displayed. However, the clustering nature of these outliers tells valuable information. Therefore, it is advised that outliers be retained in the analysis to reflect the true variability inherent in the intervention's effects.
```

## Formal analysis

This section outlines general strategies for quantifying and comparing the impacts of five different interventions on the consumption of zero-calorie and sugary beverages across three sites.

> It is recommended to implement a Generalized Linear Mixed Model (GLMM) to address the problems. A GLMM, an extension of the Linear Mixed Model (LMM) [@winter2013linear], can incorporate both fixed and random effects, and also accommodate the response variable being a count variable through a link function [@dobson2018introduction]. Specifically, while the fixed effects assess the association of covariates with the response variable across the overall population---serving as a baseline---the random effects account for variations in some associations across different entities or individuals.

Reviewer: Here I'd like to see the reasons about why the nature of problem and dataset match the GLMM, and what kind of assumptions of GLMM we need to check before implementing it to the dataset.

Reviewer 3: Why should "total" be a fixed effect?

> In this study, the primary focus is to analyze the effects of the interventions, and therefore, it is suggested to treat the intervention as a fixed effect. Additionally, days of the week and total daily beverage consumption should also be considered as fixed effects because they apply to all three sites. To allow for a more comprehensive understanding of the intervention effects, it is advised to include sites as a random effect in the model. These covariates control for variability that might influence the patterns of zero-calorie/sugary beverage consumption. For instance, site differences (which are the secondary statistical question) may impact purchasing behavior due to demographic variations between urban and suburban areas. Similarly, accounting for total daily beverage consumption helps isolate the effect of human traffic on zero-calorie/sugary beverage choices. This comprehensive approach ensures that the analysis accurately reflects the interventions' impact while controlling for external factors. Last but not least, given that the consumption of ZeroCal beverage is a non-negative integer, using an exponential link function to guarantee a positive output is suggested.

Reviewer: which **is** the secondary statistical question

Reviewer 3: * Why exponential link? You mention Poisson later.
* Sure non-negative, but what else?

Reviewer 3: > This comprehensive approach ensures that the analysis accurately reflects the interventions' impact while controlling for external factors.

I don't understand what this means. Seems like nonsense? Is it statistical?

> In this study, the primary focus is to analyze the effects of the interventions, and therefore, it is suggested to treat the intervention as a fixed effect. Additionally, days of the week and total daily beverage consumption should also be considered as fixed effects because they apply to all three sites. To allow for a more comprehensive understanding of the intervention effects, it is advised to include sites as a random effect in the model. These covariates control for variability that might influence the patterns of zero-calorie/sugary beverage consumption. For instance, site differences (which are the secondary statistical question) may impact purchasing behavior due to demographic variations between urban and suburban areas. Similarly, accounting for total daily beverage consumption helps isolate the effect of human traffic on zero-calorie/sugary beverage choices. This comprehensive approach ensures that the analysis accurately reflects the interventions' impact while controlling for external factors. Last but not least, given that the consumption of ZeroCal beverage is a non-negative integer, using an exponential link function to guarantee a positive output is suggested.

Reviewer 2: > "...For instance, site differences **(which are the secondary statistical question)** may impact..." - For instance, site differences, **the focus of our secondary statistical question**, may impact

> "....using an exponential link function....." - You might want to specify if exponential link function as being typically associated with Poisson/negative binomial regression models for count data would be applied here.

> "Similarly, accounting for **total daily beverage consumption** helps isolate the effect of **human traffic** on zero-calorie/sugary beverage choices." - For better understanding, you might want to clarify better how **total daily beverage consumption** is associated with **human traffic**.

Hence, the proposed models are \begin{align*}
ZeroCal & \sim Intervention(Site) + DofW + Total\\
Sugary & \sim Intervention(Site) + DofW + Total.

> \end{align*} Fixed effects include Intervention, DofW, and Total, while the random effect measures how the effect of Intervention and baseline consumption varies across sites. Note that these models are simplified versions, especially with the omission of the link function. Please refer to the appendix for the formulated models.

Reviewer: How to interpret the fixed the random effects in this model? Proposing a specific example could be helpful.

Reviewer 3: * Those equations are ugly. Text in equations needs to be handled differently. 
* What's `DofW`. Sure that's a column in the data. But better to use good notation or abbreviations than mimic some ugly data handling.

However, there are several limitations to the models mentioned above. Firstly, GLLMs do not accommodate autocorrelation in time series data, which refers to the dependence between consecutive observations. Moreover, the model proposed implicitly assumes that the response variables follow Poisson distributions [@coxe2009analysis], where the expectation and variability of response variables should equate, but this assumption can be easily violated in practice.

## Conclusions

> We recommend using GLMMs to fit the relationship between covariates and response variables. Based on different research interests, response variable should be either ZeroCal beverage consumption or sugary beverage consumption. The fixed-effect explanatory variables suggested are intervention, Total and DofW, and the random-effect explanatory variable is Site.

Reviewer: This part looks too general to me. Maybe you can consider add some content about how to do EDA, and also you mainly conclude how to develop the model, I suggest talking something more about how to interpret the concerned statistical questions according to the outputs of the model.

## References

::: {#refs}
:::

## Statistical Appendix

### Mathematical Formulation of Models

Here is a formulation of the Generalized Linear Mixed Model within the context of this research. As suggested earlier, random effects can be placed on both the intervention's slope and intercept. Taking $response = ZeroCal$ as an example, the model will take the format of $$

> E(zero\_calorie_{ij}) = exp((\beta_0 + b_{0i}) + (\beta_1 + b_{1i})\; Intervention + \beta_2 \; DofW + \beta_3 \; Total)

Reviewer 3: Again, no text, especially with underscores, should be used as variables in math equations.
